### PR TITLE
Change labels and transitions attr name, etc

### PR DESCRIPTION
**Pull request by andreif**
...use enum instead of int in field value; raise validation error when enum is not found by value or name (and support digit str), etc

WARNING: Inconsistent field value atm -- enum for Django 1.8 and int for other versions when model is initialized

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
   - 3.4
+  - 3.5
 
 env:
-  - DJANGO="Django>=1.5,<1.6"
-  - DJANGO="Django>=1.6,<1.7"
-  - DJANGO="Django>=1.7,<1.8"
-  - DJANGO="Django>=1.8,<1.9"
+  - DJANGO=1.5
+  - DJANGO=1.6
```
Comment (fcurella) [inline .travis.yml:12]: i think it would be totally fine to drop django 1.5 and django 1.6

Comment (andreif) [inline .travis.yml:12]: We have some projects still running on Django 1.1 :fearful: Not sure where the border is

**Comment by fcurella** [comment]
I'm not entirely convinced we should worry that much about backward-compatibility -- as long as we version correctly (ie: increment the Major version) and provide a documented upgrade path

File: .travis.yml [hunk 2]
```diff
@@ -1,26 +1,47 @@
...
+  include:
+    - python: 3.4
+      env: DJANGO=1.8  COVERAGE=true  COVERALLS_REPO_TOKEN=LdECqqwg7eelQx9w8gvooUZCFIaCqGZCv
+  allow_failures:
+    - env: COVERAGE=true
 
 install:
   - pip install flake8
-  - pip install -q $DJANGO
+  - pip install -q "Django>=$DJANGO,<$DJANGO.99"
```
Comment (fcurella) [inline .travis.yml:30]: I would keep the env vars specified as they currently are, just in case there's going to be a `1.9.99`

File: .travis.yml [hunk 3]
```diff
@@ -1,26 +1,47 @@
...
   - 3.3
   - 3.4
+  - 3.5
 
 env:
-  - DJANGO="Django>=1.5,<1.6"
-  - DJANGO="Django>=1.6,<1.7"
-  - DJANGO="Django>=1.7,<1.8"
-  - DJANGO="Django>=1.8,<1.9"
+  - DJANGO=1.6
```
Comment (fcurella) [inline .travis.yml:12]: django 1.6 is not supported anymore. we could drop this as well. Possibly we can add `1.9a1`

**Comment by fcurella** [comment]
>  enum for Django 1.8 and int for other versions

I can't find this in the diff. Can you point me to the right line?

**Comment by andreif** [comment]
Here you go https://github.com/fcurella/django-enumfield/pull/2/files#diff-174b02e9ae9348ecc51bb69323658307R48

Just to be clear, the issue is the result of my changes

File: .travis.yml [hunk 4]
```diff
@@ -1,26 +1,47 @@
...
+  - DJANGO=1.7
+  - DJANGO=1.8
+
+matrix:
+  exclude:
+    - python: 3.5
+      env: DJANGO=1.6
+    - python: 3.5
+      env: DJANGO=1.7
+  include:
```
Comment (fcurella) [inline .travis.yml:22]: I'm not sure why this `include` is needed. Isn't this ocmbo already covered by `python` \* `env`?

File: .travis.yml (hunk 2 above)
Comment (andreif) [inline .travis.yml:30]: We can always pad another 9 :smiley_cat:

File: .travis.yml (hunk 4 above)
Comment (andreif) [inline .travis.yml:22]: Yeah, annoying thing, but it's needed due to `allow_failures`

File: .travis.yml (hunk 2 above)
Comment (fcurella) [inline .travis.yml:30]: Does the `DJANGO="Django>=1.8,<1.9"` have any drawbacks? I'd rather have something consistently reliable, instead of a possible time bomb to patch in the future

File: .travis.yml (hunk 4 above)
Comment (fcurella) [inline .travis.yml:22]: oh i see

File: .travis.yml (hunk 2 above)
Comment (andreif) [inline .travis.yml:30]: It's harder to read and exclude. Will the shorter version fail in some cases?

File: django_enumfield/tests/test_enum.py [hunk 5]
```diff
@@ -32,16 +33,19 @@ def test_enum_field_save(self):
...
         self.assertEqual(person.status, PersonStatus.ALIVE)
         person.status = PersonStatus.DEAD
         person.save()
+        self.assertTrue(isinstance(person.status, PersonStatus))
         self.assertEqual(person.status, PersonStatus.DEAD)
 
         person = Person.objects.get(pk=pk)
         self.assertEqual(person.status, PersonStatus.DEAD)
+        # TODO: find a way to get only one type
+        self.assertTrue(isinstance(person.status, (PersonStatus, int)))
```
Comment (fcurella) [inline django_enumfield/tests/test_enum.py]: If I'm reading this right, the `int` vs `enum` inconsitency is independent from the version of Django.

It's simply that you set an `enum` on `save`, but you're getting an `int` back when you fetch the object. Can you confirm I'm reading this right?

File: .travis.yml (hunk 3 above)
Comment (andreif) [inline .travis.yml:12]: 1.5 was annoying due to tests import but 1.6 is fine as far as I can see. Let's wait if there are no coding complications with 1.6

File: .travis.yml (hunk 2 above)
Comment (fcurella) [inline .travis.yml:30]: A correct statement is that the shorter version it will be incorrect in some edge cases, so it's really a perfectionist issue. I'm fine either way, I don't want this to become a distracting bikeshed

File: django_enumfield/tests/test_enum.py (hunk 5 above)
Comment (andreif) [inline django_enumfield/tests/test_enum.py]: Yeah, 1.8 does not update the field value after save, but 1.7 and older update to int

Comment (andreif) [inline django_enumfield/tests/test_enum.py]: ops, that was not correct

Comment (andreif) [inline django_enumfield/tests/test_enum.py]: Django 1.8 seem to convert db-to-python value, but 1.7 and older do not.

Comment (fcurella) [inline django_enumfield/tests/test_enum.py]: Crazy Idea: would be possible to have an `enum` that could pass the `isinstance` check for `int` _and_ for `enum`?

Comment (andreif) [inline django_enumfield/tests/test_enum.py]: I was looking into wrong line for a while trying to figure out what was going on and that is still sitting in my head :smiley_cat: Need to take a break

Comment (fcurella) [inline django_enumfield/tests/test_enum.py]: feel free to take a break, I'll look into hacking `isinstance`

Comment (andreif) [inline django_enumfield/tests/test_enum.py]: I am quite tired atm so all I can think of right now is that we should probably avoid it but maybe tomorrow I'll think differently. It won't hurt to try anyway

Comment (fcurella) [inline django_enumfield/tests/test_enum.py]: I think it should work if we just subclass from `enum.IntEnum` instead of from `enum.Enum`

Comment (andreif) [inline django_enumfield/tests/test_enum.py]: It doesn't help unfortunately, however it may be a good idea to subclass `IntEnum` anyway since we currently work with integers only. In future, we will add support for string enums and it may be time to start making some preparations for it as well so we have an easy structure to extend.

**Comment by andreif** [comment]
I've tried `SubfieldBase` and according to [the docs](https://docs.djangoproject.com/en/1.6/howto/custom-model-fields/#the-subfieldbase-metaclass) it should have helped as far as I understand, but it did not. I am putting debugging on hold atm and will come back to it when I have more time.

**Comment by andreif** [comment]
We are discussing it internally and looking into dropping support for Django 1.6-1.7 to avoid inconsistency and therefore making major release. What do you think about it?

**Comment by fcurella** [comment]
Django 1.7 is still officially supported by the Django project, I would advise to continue supporting it until 1.9 is out of beta. I think Django 1.6 can be safely dropped as it's unsupported now.

File: django_enumfield/db/fields.py [hunk 6]
```diff
@@ -54,6 +64,10 @@ def from_db_value(self, value, expression, connection, context):
 
         return value
 
+    def to_python(self, value):
+        if value is not None:
```
Comment (fcurella) [inline django_enumfield/db/fields.py:68]: to_python needs to handle the following cases:
- `value` is `None`
- `value` is a string
- `value` is an enum

ref https://docs.djangoproject.com/en/1.8/howto/custom-model-fields/#converting-values-to-python-objects

Comment (andreif) [inline django_enumfield/db/fields.py:68]: `enum.get` does it

File: django_enumfield/enum.py [hunk 7]
```diff
@@ -23,6 +24,9 @@ def label(self):
 class Enum(NativeEnum):
```
Comment (fcurella) [inline django_enumfield/enum.py]: I've tried subclassing from `IntEnum` and the tests pass, regardless of the version: https://travis-ci.org/5monkeys/django-enumfield/builds/86409961

Comment (fcurella) [inline django_enumfield/enum.py]: hold on, that PR is targeting the wrong branch

Comment (fcurella) [inline django_enumfield/enum.py]: fixed link https://travis-ci.org/5monkeys/django-enumfield/builds/86411118

**Comment by fcurella** [comment]
It took some work to track it down, but I think I got `IntEnum` to work: https://travis-ci.org/5monkeys/django-enumfield/builds/86424033 https://github.com/5monkeys/django-enumfield/pull/28

**Comment by andreif** [comment]
Awesome!

**Comment by fcurella** [comment]
Looking good :)

I would add a Changelog (either its own file or a section in the readme) listing the breaking changes:
- `django_enumfield.enum.Enum` is now a subclass of the native `IntEnum`
- `_transitions` is now called `__transitions__`
- `_labels` is now called `__labels__`
